tests/Unit/Evolution/DiscontinuousGalerkin/Test_InboxTags.cpp

```diff
@@ -0,0 +1,180 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <cstddef>
+#include <random>
+#include <vector>
+
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/DiscontinuousGalerkin/InboxTags.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "Helpers/DataStructures/DataBox/TestHelpers.hpp"
+#include "Helpers/DataStructures/MakeWithRandomValues.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "Time/TimeStepId.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace evolution::dg {
+namespace {
+template <size_t Dim>
+void test_no_ghost_cells() {
+  static constexpr size_t number_of_components = 1 + Dim;
+  using bc_tag = Tags::BoundaryCorrectionAndGhostCellsInbox<Dim>;
+  using Type = std::tuple<Mesh<Dim - 1>, std::optional<std::vector<double>>,
+                          std::optional<std::vector<double>>, ::TimeStepId>;
+  using Inbox = typename bc_tag::type;
+
+  std::uniform_real_distribution<double> dist(-1.0, 2.3);
+  MAKE_GENERATOR(gen);
+
+  const TimeStepId time_step_id_a{true, 3, Time{Slab{0.2, 3.4}, {3, 100}}};
+  const TimeStepId time_step_id_b{true, 4, Time{Slab{3.4, 5.4}, {13, 100}}};
+  const TimeStepId time_step_id_c{true, 5, Time{Slab{5.4, 6.4}, {17, 100}}};
+  const std::pair nhbr_key{Direction<Dim>::lower_xi(), ElementId<Dim>{1}};
+
+  Inbox inbox{};
+
+  Type send_data_a{};
+  get<0>(send_data_a) = Mesh<Dim - 1>{5, Spectral::Basis::Legendre,
+                                      Spectral::Quadrature::GaussLobatto};
+  get<2>(send_data_a) = std::vector<double>(
+      get<0>(send_data_a).number_of_grid_points() * number_of_components);
+  get<3>(send_data_a) = time_step_id_a;
+  fill_with_random_values(make_not_null(&*get<2>(send_data_a)),
+                          make_not_null(&gen), make_not_null(&dist));
+
+  bc_tag::insert_into_inbox(make_not_null(&inbox), time_step_id_a,
+                            std::make_pair(nhbr_key, send_data_a));
+
+  CHECK((inbox.at(time_step_id_a).at(nhbr_key) == send_data_a));
+
+  Type send_data_b{};
+  get<0>(send_data_b) = Mesh<Dim - 1>{7, Spectral::Basis::Legendre,
+                                      Spectral::Quadrature::GaussLobatto};
+
+  get<2>(send_data_b) = std::vector<double>(
+      get<0>(send_data_b).number_of_grid_points() * number_of_components);
+  get<3>(send_data_b) = time_step_id_c;
```

Comment:
As far as I can tell, this surprising ("blah_b = blah_c") line isn't testing something new, the way it is in the ghost zone test case below. I'd suggest just using time_step_id_b here to minimize surprise

Reply:
I've added a comment about what it tests.